teps/0127-larger-results-via-sidecar-logs.md

```diff
@@ -0,0 +1,293 @@
+- [Experiment Questions](#experiment-questions)
+- [References](#references)
+<!-- /toc -->
+
+This TEP builds on the hard work of many people who have been tackling the problem over the past couple years,
+including but not limited to:
+- '@abayer'
+- '@afrittoli'
+- '@bobcatfish'
+- '@dibyom'
+- '@imjasonh'
+- '@pritidesai'
+- '@ScrapCodes'
+- '@skaegi'
+- '@tlawrie'
+- '@tomcli'
+- '@vdemeester'
+- '@wlynch'
+
+## Summary
+
+Today, `Results` have a size limit of 4KB per `Step` and 12KB per `TaskRun` in the best case - see [issue][4012]. 
+
+The goal of [TEP-0086][tep-0086] is to support larger `Results` beyond the current size limits. TEP-0086 has many
+alternatives but no proposal. This TEP proposes experimenting with one of the alternatives - `Sidecar` logs. This
+allows us to support larger `Results` which are stored within `TaskRun` CRDs.
+
+## Motivation
+
+`Results` are too small - see [issue][4012]. The current implementation of `Results` involves parsing from disk and
+storing as part of the `Termination Message` which has a limit of 4KB per `Container` and 12KB per `Pod`. As such,
+the size limit of `Results` is 12KB per `TaskRun` and 4KB per `Step` at best.
+
+To make matters worse, the limit is divided equally among all `Containers` in a `Pod` - see [issue][4808]. The more
+the `Steps` in a `Task`, the less the size limit for `Results`. For example, if there are 12 `Steps` then each has
+1KB in `Termination Message` storage to produce `Results`.
+
+[TEP-0086][tep-0086] aims to support larger `Results`. It has many [alternatives][tep-0086-alts] but no proposal
+because there's no obvious "best" solution that would meet all the requirements.
+
+This TEP proposes experimenting with `Sidecar` logs to support larger `Results` that are stored within `TaskRun` CRDs.
+This allows us to provide an immediate solution to the urgent needs of users, while not blocking pursuit of the other
+alternatives.
+
+In addition, the [documented guidance][docs] is that `Results` are used for outputs less than 1KB while `Workspaces`
+are used for larger data. Supporting larger `Results` up to the CRD limit allows users to reuse `Tasks` in more
+scenarios without having to change the specification to use `Workspaces` upon hitting the current low size limit of
+4KB per `TaskRun`.
+
+> As a general rule-of-thumb, if a `Result` needs to be larger than a kilobyte, you should likely use a `Workspace`
+to store and pass it between `Tasks` within a `Pipeline`.
+
+### Goals
+
+The main goal of this TEP is to support larger `Results` via `Sidecar` logs. The `Results` are stored in the `TaskRun`,
+therefore they are limited by the size limit of a `TaskRun` CRD - 1.5MB. 
+
+### Non-Goals
+
+The following are out of scope for this TEP:
+
+1. Solving use cases that requires really large `Results` beyond the size limit of a CRD - 1.5MB.
+
+2. Addressing other [alternatives][tep-0086-alts] for larger `Results` that are listed in [TEP-0086][tep-0086].
```

Comment:
This is fine of course, but I wonder if we should include in the goals proposing a solution that may cohexists with other solutions from the alternatives in TEP-0086, for instance via feature flag to being with.

Reply:
Agreed that they should coexist, updated the TEP